node/src/components/linear_chain_sync/operations.rs

```diff
@@ -263,16 +269,120 @@ where
     return Ok(outstanding_tries);
 }
 
+struct BlockWithMetadataFetchedFromRemote<I> {
+    block_with_metadata: Box<BlockWithMetadata>,
+    sender: I,
+}
+
+/// Fetches a block from the network by height.
+async fn fetch_and_store_block_by_height_from_remote<REv, I>(
```

Comment:
Maybe:
```suggestion
async fn fetch_and_store_block_by_height<REv, I>(
```
Because it's not necessarily from remote, if we already have it in storage?

Reply:
I'm not sure.  What happens if I try to validate this with the `block_validator` and I don't have the right sender?

Reply:
Ah, you mean then it will conclude that nobody has the deploys? :confused:

Reply:
I made a follow up ticket to remedy this.